Server/datalayer/user.datalayer.js

```diff
@@ -8,6 +10,54 @@ class usersDatalayer {
     * Initializes the datalayer
     */
 
+    // Creates new user 
+    static createUser(userObj) {
+        bcrypt.hash(userObj.password, saltRounds).then(function(hashPassword) {
+            let user = new userModel(
+                {
+                    username: userObj.username,
+                    email: userObj.email,
+                    password: hashPassword,
+                    firstname: userObj.firstname,
+                    lastname: userObj.lastname
+                });
+                return user.save();
+        }).catch(err => {
+            throw new error(`Not able to save - ${err}`);
+        });
+    }
+
+    // Get user details based on id
+    static getUserById(userid) {
+        return userModel.findById(userid).exec().catch(err => {
+            throw new Error(`Not able to get user - ${err}`);
+        });
+    }
+
+    // Update user details based on id
+    static updateUserById(userid , updateUserObj) {
+        bcrypt.hash(updateUserObj.password, saltRounds).then(function(hashPassword) {
```

Comment:
what if user is not updating password?
then `updateUserObj.password` will be undefined, for now not lets assume user will not update password

Reply:
Updated as suggested . Now user can update any field except password.